Android/app/build.gradle

```diff
@@ -15,6 +15,18 @@ try {
     keystoreProperties['storePassword'] = ''
 }
 
+// Go backend build constants
+def goToolChain = 'go1.21.1'
```

Comment:
We should use whatever is in the go.mod. 
Please add a comment.

I would expect and newer Go to pull the right version from the Go mod, as we saw in the 1.21 release notes.

Reply:
This would work if you call `go build` directly, but it seems `gomobile` doesn't recognize the toolchain version in `go.mod`. Therefore I used the environment variable here to pass the toolchain version all the way down to the `go` compiler.

Reply:
But I can read the version out from `go.mod`, I updated the build script and added the comment.